### PR TITLE
fix close ping message

### PR DESCRIPTION
**Pull request by frankffenn**

### CODE REVIEW

File: websocket.go [hunk 1]
```diff
@@ -531,7 +533,9 @@ func (c *wsConn) handleWsConn(ctx context.Context) {
 	// setup pings
 
 	stopPings := c.setupPings()
-	defer stopPings()
+	defer func() {
+		stopPings()
```
Comment (Stebalien) [inline websocket.go:537]: So, this code _was_ incorrect. However, it's now racy. Before, `stopPings` was:

1. Read exactly once here.
2. Written from within a goroutine, under the write lock.

Now, it's read when this function returns, when the write lock isn't held.

Comment (frankffenn) [reply websocket.go:537]: When the `stopPings`  is re-assigned in `c.incoming` case , `stop` channel  cannot be closed properly

Comment (Stebalien) [reply websocket.go:537]: I know, I'm just saying that this patch isn't the right fix.

**Comment by Stebalien** [comment]
Ok, I looked into fixing this correctly, but I have no idea how this code is supposed to work. I'll get back to you on that.

**Comment by BigOceanGG** [comment]
@Stebalien @frankffenn Has this problem been solved？

**Comment by jennijuju** [comment]
may fix https://github.com/filecoin-project/lotus/issues/3597

**Pull request closed without merging**